mlflow/store/artifact/databricks_models_artifact_repo.py

```diff
@@ -110,7 +111,8 @@ def _get_signed_download_uri(self, path=None):
                 "API request to get presigned uri to for file under path `%s` failed with"
                 " status code %s. Response body: %s" % (path, response.status_code, response.text)
             )
-        return json_response.get("signed_uri", None), json_response.get("headers", None)
+        # We must default the headers to be {} as that is the type expected back in the consumers
+        return json_response.get("signed_uri", None), json_response.get("headers", {})
```

Comment:
In terms of how to fix model_artifact_repo parsing: It looks like we need to replace this code block https://github.com/mlflow/mlflow/blob/b08aa03ed35f76ee143022ac2d535576fecfaeb2/mlflow/store/artifact/databricks_models_artifact_repo.py#L105-L113
Instead of using the raw json_response, we will use the utils here https://github.com/mlflow/mlflow/blob/b08aa03ed35f76ee143022ac2d535576fecfaeb2/mlflow/utils/rest_utils.py#L262-L277

Reply:
Agree.  That's a slightly scarier change since this call_endpoint is used in a few places that we would want to test if we were reworking it completely.  I'm inclined to land this fix and in a follow up commit deal with this.